app/controllers/api/v1/comments_controller.rb

```diff
@@ -0,0 +1,35 @@
+class Api::V1::CommentsController < ApplicationController
+  skip_before_action :authenticate_user!
+  protect_from_forgery unless: -> { request.format.json? }
+  def index
+    @post = Post.find(params[:post_id])
+    @comments = Comment.where(post: @post)
+    render json: @comments
```

Comment:
- You have done well so far, but according to the project requirement, you are to create an API endpoint to list all comments for a user's post, from your routes for this action, `/api/v1/users/:user_id/posts/:post_id/comments`, we need to find the user using the user_id, then find the specific post belonging to that user. After finding the post we can get the comments belonging to that post. Please fix this.
```ruby
     @post = Post.find_by(user_id:params[:user_id], post_id:params[:post_id])
     @comments = @posts.comments  
```